docs/guides/src/main/server/run-containers.adoc

```diff
@@ -0,0 +1,106 @@
+
+== Create an optimized container image
+To get the best startup experience for your Keycloak container, we recommend building an optimized container image by running the `build` step explicitly before starting:
+
+=== Build your optimized Keycloak docker image
+The `Dockerfile` below creates a pre-configured Keycloak image which enables the metrics endpoint, the token exchange feature and uses a postgres database.
+
+.Dockerfile:
+[source, dockerfile]
+----
+FROM quay.io/keycloak/keycloak-x:latest as builder
+
+ENV KC_METRICS_ENABLED=true
+ENV KC_FEATURES=token_exchange
+ENV KC_DB=postgres
+RUN /opt/keycloak/bin/kc.sh build
+
+FROM quay.io/keycloak/keycloak-x:latest
+COPY --from=builder /opt/keycloak/lib/quarkus/ /opt/keycloak/lib/quarkus/
+WORKDIR /opt/keycloak
+# for demonstration purposes only, please make sure to use proper certificates in production instead
+RUN keytool -genkeypair -storepass password -storetype PKCS12 -keyalg RSA -keysize 2048 -dname "CN=server" -alias server -ext "SAN:c=DNS:localhost,IP:127.0.0.1" -keystore conf/server.keystore
+ENV KEYCLOAK_ADMIN=admin
+ENV KEYCLOAK_ADMIN_PASSWORD=change_me
+# change these values to point to a running postgres instance
+ENV KC_DB_URL=<DBURL>
+ENV KC_DB_USERNAME=<DBUSERNAME>
+ENV KC_DB_PASSWORD=<DBPASSWORD>
+ENV KC_HOSTNAME=localhost:8443
+ENTRYPOINT ["/opt/keycloak/bin/kc.sh", "start"]
+----
+In the first stage of this multi-staged build, we're creating the optimized image using the `build` command to apply the build options. From the `builder`, we're copying the files generated by the `build` process into a new image. In this runner image, we apply the specific run configuration, containing e.g. a keystore, the environment-specific hostname configuration and database configuration. Then this image gets started in production mode by using the `start` command in the entrypoint.
+
+Note that this example uses a multi-staged build to explicitly showcase the build and run steps. It could also be run as a single-staged docker build.
+
+=== Building the docker image
+To build the actual docker image, run the following command form the Directory containing your Dockerfile:
+[source,bash]
+----
+podman|docker build . -t prebuilt_keycloak
+----
+
+=== Start the optimized Keycloak docker image:
+To start the image, run:
+[source, bash]
+----
+podman|docker run --rm -it --name optimized_keycloak -p 8443:8443 prebuilt_keycloak
+----
+
+You'll notice that the startup log contains the following line:
+[source, bash]
+----
+INFO  [org.key.com.Profile] (main) Preview feature enabled: token_exchange
+----
+stating that the desired feature is enabled.
+Also, opening up `https://localhost:8443/metrics` leads to a page containing operational metrics which could be used by your monitoring solution.
```

Comment:
Probably better to avoid `Also,`
```suggestion

Opening up `https://localhost:8443/metrics` leads to a page containing operational metrics which could be used by your monitoring solution.
```